### PR TITLE
fix: resolve Ruff RET504 warnings in feature, filters, and geometry

### PR DESCRIPTION
**Pull request by Futtano**
## What does this pull request do?

Resolves Ruff RET504 warnings in `kornia/feature`, `kornia/filters`, and `kornia/geometry` by returning expressions directly instead of assigning them to temporary variables immediately before return. The changes preserve existing behavior and continue the return-rule cleanup tracked in #2445.

Each package is kept in a separate commit to make review and rollback straightforward. RET504 remains disabled globally because 74 violations still exist elsewhere in `kornia`, primarily under `kornia/models`. Once those are resolved, the individually selected RET rules can be replaced with plain `RET` to enforce the complete family.

This is an internal cleanup with no intended public behavior change. No changelog fragment is included; please apply the `no-changelog` label.

## Related issue or discussion

Refs #2445

## What did you check?

- The complete RET rule family passes in all three changed packages.
- Feature tests: 829 passed, 98 skipped.
- Filters tests: 1,161 passed, 10 skipped.
- Geometry tests: 3,113 passed, 80 skipped, 10 xfailed, 6 xpassed.
- `pixi run pre-commit-all` passes.
- `pixi run typecheck` passes.
- AST comparisons confirmed that all 57 modified files are equivalent after normalizing assignments that are immediately returned.

```text
$ pixi run uv run ruff check --select RET kornia/feature kornia/filters kornia/geometry
All checks passed!

$ pixi run pre-commit-all
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
check yaml...............................................................Passed
check toml...............................................................Passed
check for merge conflicts................................................Passed
check for added large files..............................................Passed
mixed line ending........................................................Passed
pretty format json.......................................................Passed
ruff check...............................................................Passed
ruff format..............................................................Passed
codespell................................................................Passed
Add license header.......................................................Passed

$ pixi run typecheck
All checks passed!
```

## How did AI help?

I directed the package-by-package cleanup and reviewed the resulting changes. Codex applied the RET504 refactors, checked semantic equivalence, ran the package test suites and repository checks, organized the work into focused commits, and prepared this pull request.

### CODE REVIEW

**Review by ducha-aiki — APPROVED** [review]
## Review at `192c4d9a1`

I reviewed the branch and its merge into current `main` (`8b03a8085`). The merge is clean. Nothing blocks, so I'm approving.

**No overlap with the RET504 work already merged.** #4722 touched `color`, `contrib`, `core` and `enhance`, and #4696 touched `augmentation`. This PR is limited to `feature`, `filters` and `geometry`, so there are no duplicated or conflicting hunks.

**Every rewrite preserves behavior.** I checked this independently of the PR's own AST claim. On `main` I rewrote every `x = <expr>` / `x: T = <expr>` followed by `return x` into `return <expr>`, then compared ASTs with the PR tree.
- 56 of the 57 files are identical.
- The remaining file is `filters/dissolving.py`, where `return latents * 0.18215` now sits inside the `with torch.no_grad():` block that already wrapped the computation. That is equivalent.
- No change to evaluation order, dtype, or operands.

**TorchScript.** I ran `torch.jit.script` on all 67 changed top-level functions, on `main` and on the merge.
- 66 script the same way on both trees (33 OK, 33 already unscriptable on `main`).
- The one difference is an improvement: `feature/matching.py::_get_default_fginn_params` now scripts, because the dict literal picks up the `Dict[str, Any]` return annotation. It failed on `main`.
- The `-k "jit or script"` subset of `tests/feature tests/filters tests/geometry` gives 147 passed, 37 skipped.

**Tests.** `pytest tests/feature tests/filters tests/geometry` on the merge gives 5103 passed, 188 skipped, 10 xfailed, 6 xpassed, 0 failed. `kornia.__file__` resolved to the worktree.

**No other Ruff rule comes back (ruff 0.16.8, the pre-commit pin).**
- `ruff check --isolated --select ALL` on the 57 files, `main` vs the merge: only RET504 (121 → 0) and N806 (404 → 396) change. Every other rule count is identical, so there is no RET505 → RET506 style regression.
- `ruff check --select RET` on the three packages passes.
- `ruff check kornia tests` and `ruff format --check kornia` both pass.

**Changelog.** This is an internal change, so no fragment is needed. I applied the `no-changelog` label, and the fragment check now passes. CI is 40/40 green.

Dropped (not blocking):
- `feature/adalam/utils.py::dist_matrix` now has the stale `# distmat[torch.isnan(distmat)] = np.inf` comment after the `return`. It was dead text before this PR too.
- `_nullspace_via_eigh` loses its `# orthonormal columns` note.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
